crates/ntx-builder/src/store.rs

```diff
@@ -172,21 +173,56 @@ impl StoreClient {
         Ok(all_notes)
     }
 
-    // TODO: add pagination.
+    /// Get all network account IDs.
+    ///
+    /// Since the `GetNetworkAccountIds` method is paginated, we loop through all pages until we
+    /// reach the end.
     #[instrument(target = COMPONENT, name = "store.client.get_network_account_ids", skip_all, err)]
     pub async fn get_network_account_ids(&self) -> Result<Vec<AccountId>, StoreError> {
-        let response = self.inner.clone().get_network_account_ids(()).await?.into_inner();
-
-        let accounts: Result<Vec<AccountId>, ConversionError> = response
-            .account_ids
-            .into_iter()
-            .map(|account_id| {
-                AccountId::read_from_bytes(&account_id.id)
-                    .map_err(|err| ConversionError::deserialization_error("account_id", err))
-            })
-            .collect();
-
-        Ok(accounts?)
+        const MAX_ITERATIONS: u32 = 1000;
```

Comment:
Would be great to add a comment here indicating why 1000 iterations is enough or at least state what kind of assumption we are making here about the maximum number of network accounts.